### PR TITLE
docs: add IDE integration setup instructions in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by roman-petrov**
### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### What changes did you make? (Give an overview)

I added some contributing instructions about how to run analyzer plugin in IDE from locally cloned git working copy.

### Is there anything you'd like reviewers to focus on?

I work only on Windows machines and do not have `MacOS/Linux` development experience so new instructions might be `Windows` specific. My English might be not good enough :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -59,3 +61,23 @@ To create a new rule:
...
+2. Change `dart_code_metrics` dependency in `tools\analyzer_plugin\pubspec.yaml` to path dependency:
+
+    ```yaml
+    dependencies:
+    dart_code_metrics:
+      path: <absolute-path>
+    ```
+
+3. Do the same in your project(s) you wish to work on `dart-code-metrics`: reference it from absolute path.
+4. For Visual Studio Code on Windows: delete `C:\Users\<your-windows-user-name>\AppData\Local\.dartServer` folder.
```
Comment (roman-petrov) [inline CONTRIBUTING.md:82]: Do we need instructions for other platforms/IDEs here right now or we might add them in future?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,3 +61,23 @@ To create a new rule:
...
+1. Assume you have cloned the repository into `<absolute-path>` directory.
+2. Change `dart_code_metrics` dependency in `tools\analyzer_plugin\pubspec.yaml` to path dependency:
+
+    ```yaml
+    dependencies:
+    dart_code_metrics:
+      path: <absolute-path>
+    ```
+
+3. Do the same in your project(s) you wish to work on `dart-code-metrics`: reference it from absolute path.
```
Comment (roman-petrov) [inline CONTRIBUTING.md:81]: We might need to run `dart pub get` in all places we have `pubspec.yaml`, maybe we should add additional step here...

File: CONTRIBUTING.md (hunk 1 above)
Comment (incendial) [reply CONTRIBUTING.md:82]: I think we definitely need them, but it can be done separately.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -59,3 +61,23 @@ To create a new rule:
...
 10. Add the rule tests under `test/rules/`. Prefer to split test examples to a correct/incorrect groups.
+
+## Run the plugin in IDE
+
+The plugin works in multiple IDEs and is implemented as analysis server plugin. To work on and test IDE/analysis server integration features and issues you
+will need to load the plugin into analysis server.
+
+To set this up:
+
+1. Assume you have cloned the repository into `<absolute-path>` directory.
```
Comment (incendial) [inline CONTRIBUTING.md:72]: Suggestion: Clone the repository into `<absolute-path>` directory.

I think we can simplify it (if the user had already cloned the repo, they will just skip this step).

File: CONTRIBUTING.md (hunk 1 above)
Comment (roman-petrov) [reply CONTRIBUTING.md:82]: ok, lets leave it as is for now.

**Comment by roman-petrov** [comment]
@incendial I added some changes based on comments you added, please take another look.

**Review by incendial — APPROVED** [review]
LGTM, thank you!

**Pull request merged**